### PR TITLE
Fixing logic to generate ExternalHost in genericapiserver

### PR DESCRIPTION
**Pull request by nikhiljindal**
@ncdc pointed it out (https://kubernetes.slack.com/archives/sig-api-machinery/p1464974528000139) that lines 305 and 306 dont match. We should be using ReadWritePort instead of ServiceReadWritePort in line 306.

https://github.com/kubernetes/kubernetes/pull/20626 seems to be the culprit PR.

### CODE REVIEW

**Comment by ncdc** [comment]
This is a behavioral change between 1.2 and 1.3, right? re release notes

**Comment by ncdc** [comment]
cc @kubernetes/sig-api-machinery @smarterclayton @liggitt @deads2k

**Comment by nikhiljindal** [comment]
Updated the release note label

File: pkg/genericapiserver/genericapiserver.go [hunk 1]
```diff
@@ -303,7 +303,7 @@ func setDefaults(c *Config) {
 	if len(c.ExternalHost) == 0 && c.PublicAddress != nil {
 		hostAndPort := c.PublicAddress.String()
 		if c.ReadWritePort != 0 {
```
Comment (lavalamp) [inline pkg/genericapiserver/genericapiserver.go]: This if statement will always trigger, since ReadWritePort is set on L295.

**Comment by lavalamp** [comment]
This looks correct based on the documentation for the fields.

**Comment by erictune** [comment]
@nikhiljindal  Does this PR require action by the user when upgrading from 1.2.x to 1.3.0?  (Think about non-developer users.) If so, please edit your first comment to have  a release-note block, like in https://github.com/kubernetes/kubernetes/pull/28132.  If it is just an optional feature, please change the label to just release-note.  If it is not a complete feature by itself, then apply "release-note-none" label instead.

**Pull request merged**